pyomo/core/base/block.py

```diff
@@ -1718,6 +1719,7 @@ def construct(self, data=None):
                           self.__class__.__name__, self.name, str(data) )
         if self._constructed:
             return
+        timer = ConstructionTimer(self)
```

Comment:
There is another return statement in this method (inside the if rule is None block) that needs a `timer.report()` just prior to it.